core/src/main/java/org/apache/carbondata/core/scan/filter/executer/ExcludeFilterExecuterImpl.java

```diff
@@ -146,9 +146,17 @@ private BitSet setFilterdIndexToBitSet(FixedLengthDimensionDataChunk dimColumnDa
     BitSet bitSet = new BitSet(numerOfRows);
     bitSet.flip(0, numerOfRows);
     byte[][] filterValues = dimColumnExecuterInfo.getFilterKeys();
-    for (int k = 0; k < filterValues.length; k++) {
+    if (filterValues.length > 1) {
```

Comment:
This code will not work when No Inverted index is configured for that column while creating the column. In that case data wont be sorted and this code will five wrong result and whether filter value is one or more than one this if data is sorted we can add binary search